core/src/ppu.cpp

```diff
@@ -222,4 +295,116 @@ void Ppu::clear_vblank_flag() {
+    registers_->attribute_table_shifter_hi <<= 1u;
+}
+
+void Ppu::increase_scroll_counters() {
+    if (!registers_->is_rendering_enabled()) {
+        return;
+    }
+
+    // During pixels 280 through 304 of this scanline, the vertical scroll bits
+    // are reloaded if rendering is enabled. vert(v) == vert(t)
+    if (scanline() == kPreRenderScanline) {
+        if (cycle() >= 280 && cycle() <= 304) {
+            registers_->vram_addr &= static_cast<uint16_t>(~0x7BE0u);
+            registers_->vram_addr |= (registers_->temp_vram_addr & 0x7BE0u);
+        }
+    }
+
+    const bool should_increase_coarse_x =
+            !(cycle() == 0 || (cycle() >= 256 && cycle() <= 320));
+    if (should_increase_coarse_x && (cycle() % 8) == 0) {
+        registers_->vram_addr = increase_vram_coarse_x(registers_->vram_addr);
+    }
+
+    if (cycle() == 256) {
+        registers_->vram_addr = increase_vram_y(registers_->vram_addr);
+    } else if (cycle() == 257) {
+        // Copies all bits related to horizontal position from
+        // temporal to current vram_address
+        // From http://wiki.nesdev.com/w/index.php/PPU_scrolling
+        // If rendering is enabled, the PPU copies all bits related to
+        // horizontal position from t to v:
+        registers_->vram_addr &= ~0x41Fu;
+        registers_->vram_addr |= registers_->temp_vram_addr & 0x41Fu;
+    }
+}
+void Ppu::fetch() {
+    if ((cycle() >= 1 && cycle() <= 256) ||
+            (cycle() >= 321 && cycle() <= 336)) {
+        const uint16_t background_pattern_table_base_address =
+                (registers_->ctrl & 0b0001'0000u) ? 0x1000u : 0x0000u;
+        const uint8_t fine_scroll_y = get_fine_scroll_y(registers_->vram_addr);
+
+        switch ((cycle() - 1) % 8) {
+        case 0: {
+            registers_->pattern_table_shifter_low =
+                    (registers_->pattern_table_shifter_low & 0xFF00u) |
+                    registers_->pattern_table_latch_low;
+            registers_->pattern_table_shifter_hi =
+                    (registers_->pattern_table_shifter_hi & 0xFF00u) |
+                    registers_->pattern_table_latch_hi;
+
+            registers_->attribute_table_shifter_low =
+                    (registers_->attribute_table_shifter_low & 0xFF00u) |
+                    ((registers_->attribute_table_latch & 0b01u) ? 0xFFu
+                                                                 : 0x00u);
+            registers_->attribute_table_shifter_hi =
+                    (registers_->attribute_table_shifter_hi & 0xFF00u) |
+                    ((registers_->attribute_table_latch & 0b10u) ? 0xFFu
+                                                                 : 0x00u);
+
+            registers_->name_table = registers_->name_table_latch;
+            const uint16_t nametable_address =
+                    get_nametable_address(registers_->vram_addr);
+            registers_->name_table_latch = mmu_->read_byte(nametable_address);
+        } break;
```

Comment:
The formatting of the breaks in this switch are pretty odd? Is this intentional? :O I'm not hating it, but I'm surprised clang-format doesn't care.

Reply:
I put moved break; to inside the brackets now. Is that better?

Reply:
I didn't mind it before either. I was mostly surprised clang-format didn't care. :P